x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/TrainedModelConfig.java

```diff
@@ -703,6 +704,10 @@ public Builder setHyperparameters(List<Hyperparameters> hyperparameters) {
             return addToMetadata(HYPERPARAMETERS, hyperparameters.stream().map(Hyperparameters::asMap).collect(Collectors.toList()));
         }
 
+        public Builder setPlatformArchitecture(String platformArchitecture) {
+            return addToMetadata(PLATFORM_ARCHITECTURE, platformArchitecture);
```

Comment:
A side effect of putting the architecture in the metadata rather than as a top level field is that it won't be possible to use it to search for model configs that have or don't have a particular architecture. It will only be possible to check the value after loading the full config. See: https://github.com/maxhniebergall/elasticsearch/blob/f241f2bb6ae6944799fe6cf36d5674a0ec46667f/x-pack/plugin/core/template-resources/src/main/resources/ml/inference_index_mappings.json#L34